articles/tech-resources/ios-universal-links-self-serve.md

```diff
@@ -0,0 +1,109 @@
+
+The setup steps are as follows:
+
+1. [Configure a custom tracking domain on your SparkPost account.](https://app.sparkpost.com/account/tracking-domains)
+1. [Use a "custom link sub-path" in your links.](https://developers.sparkpost.com/api/substitutions-reference.html#header-custom-link-sub-paths)
+1. Publish the universal links for your app using both the tracking domain and link sub-path. Your apple-app-site-association file might include the following:
+    ```
+    {
+      "applinks": {
+        "details": [
+          {
+            "appID": "<YOUR_APP_ID>",
+            "paths": [
+              "/f/open-in-app/*"
+            ]
+          }
+        ]
+      }
+    }
+    ```
+1. Send your email through SparkPost with click tracking enabled, using the published tracking domain and link sub-path:
+    ```
+    Your invoice is available <a data-msys-sublink="open-in-app" href="http://www.example.com/invoice/101">here</a>.
+    ```
+1. Update your iOS app to make an HTTP GET request to each universal link when it handles the incoming click event to trigger SparkPost's click tracking. More details on this below.
+1. (Optional) Have your iOS app take action based on the original URL which SparkPost returns in its "3xx redirect" HTTP response.
+
+### Forwarding Clicks From iOS To SparkPost 
+
+When an iOS email client recognises that your recipient has tapped on a universal link, it sends your application delegate an `NSUserActivity`. To allow SparkPost to track this universal link "click" event, you must visit the URL in the activity. You may also choose to use the original URL to control app behaviour.
+
+Here's a sample `application:continueUserActivity:restorationHandler` method implementation which calls the SparkPost click tracker and retrieves the original URL:
+
+```
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity
+ restorationHandler:(void (^)(NSArray *))restorationHandler {
+
+    // Ignore non-universal link activities
+    if (![userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb] || userActivity.webpageURL == nil) {
+        return NO;
+    }
+
+    // Create a task to call the SparkPost click tracker, expecting a 3xx redirect to your original URL.
+    // On completion, the handler will receive an NSURLResponse which knows the original target URL.
+    NSURLSessionDataTask* clickTask = [[NSURLSession sharedSession] dataTaskWithURL:userActivity.webpageURL completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+        if (response) {
+            NSURL* originalURL = [response URL];
+            if (originalURL) {
+                NSLog(@"Target URL -> %@", originalURL.absoluteString);
+            }
+        }
+    }];
+
+    // Kick off the task
+    [clickTask resume];
+
+    return YES;
+}
+```
```

Comment:
I have no idea if we do color coding, but if we do, adding objc (or whatever the keyword is) after the first three backticks might make it work.